server/src/main/java/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -547,14 +547,12 @@ public DomainRouterVO deployRouter(final RouterDeploymentDefinition routerDeploy
                 reallocateRouterNetworks(routerDeploymentDefinition, router, template, null);
                 router = _routerDao.findById(router.getId());
             } catch (final InsufficientCapacityException ex) {
-                if (allocateRetry < 2 && iter.hasNext()) {
+                if (iter.hasNext()) {
```

Comment:
could it cause regression, where earlier for one hypervisor it was trying at least twice?

Reply:
@rohityadavcloud 
`allocateRetry < 2` means that it will try with 2 hypervisors, not try twice for each hypervisor.

Reply:
ah :) my bad - LGTM thnx @weizhouapache